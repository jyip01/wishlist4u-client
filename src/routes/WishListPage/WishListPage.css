.Button__AddList{
    background-color: rgba(228, 228, 250, 0.831);
    border: 1px solid rgba(31, 31, 41, 0.824);
    border-radius: 6px;
    color: rgb(37, 35, 58);
    font-size: 13px;
    margin-top: 10px;
    font-weight: bold;
  }

.WishListPage__Landing{
  background-color: rgba(238, 230, 255, 0.577);
  border-bottom: 1px solid black;
  padding: 5px;
}

.WishListPage__Landing p{
  color: black;
  text-align: center;
  font-size: 15px;
}

.WishListPage__ButtonAndFormContainer{
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.WishListPage__Landing-demo-credentials{
  text-align: center;
  list-style-type: none;
  font-weight: bold;
  color: black;
  text-shadow: 2px 2px 2px cornflowerblue;
}

.WishListPage__Landing-header{
  text-align: center;
  text-decoration: underline cornflowerblue;
  text-decoration-style: wavy;
  font-style: italic;
  font-weight: bold;
}

.WishListPage{
  position: relative;
  
  width: 100%;
  margin-left: 0;
  margin-top: 0;
}

.WishListPage:hover{
  position: relative;
  top: 5px;
}

.signinLink{
  color: red;
  text-shadow: 0 0 1px black;
}

.signinLink:hover{
  color: cornflowerblue;
  text-shadow: 0 0 3px black;
  font-size: 16px;
  font-weight: bold;
}

.AddPlaceHolder{
  text-align: center;
  font-weight: 400;
  color: black;
  text-shadow: rgb(238, 230, 255);
}