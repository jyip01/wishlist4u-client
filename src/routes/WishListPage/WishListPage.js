import React, { Component } from "react";
import WishListContext from "../../contexts/WishListContext";
import ListApiService from "../../services/list-api-service";
import { Section, Button } from "../../components/Utils/Utils";
import ListWishList from "../../components/ListWishList/ListWishList";
import ListForm from "../../components/ListForm/ListForm";
import TokenService from '../../services/token-service';
import './WishListPage.css';
import { Link } from 'react-router-dom';


export default class WishListPage extends Component {
  state = {
    add: false,
  };

  static contextType = WishListContext;

  componentDidMount() {
    this.context.clearError();
    if(TokenService.hasAuthToken()){
      const id = TokenService.readJwtToken().user_id

      ListApiService.getLists(id)
      .then(this.context.setListWishList)
      .catch(this.context.setError);
    }else{
      this.props.history.push('/login')
    }
    
  }

  landingPageLogin = () => {
    if (!TokenService.hasAuthToken()) 
      return (
        <div className='WishListPage__Landing'>
          <h2 className='WishListPage__Landing-header'>Welcome to WishList4U</h2>
          <p>Need help keeping track of all your wants and needs AS WELL as your friends and family? Look no further, WishList4U will help you keep track of all that!</p>
          <p>We've added some lists to get you started and give you an idea of how to use it, before adding your own list or making new wishes, please Login or Create an Account.</p>
          <p> If you're just here to demo the app please use the credentials below:</p>
          <ul className='WishListPage__Landing-demo-credentials'>
            <li>User Name: Test_User</li>
            <li>Password: P@ssword1</li>
          </ul>
        </div>
      );
    else
      return (<> </>);
  }

  addListButtonHandler = () => {
    !this.state.add
      ? this.setState({ add: true })
      : this.setState({ add: false });
  };

  addedList = () => {
    this.setState({add: false})
  }

  refreshLists = () => {
    ListApiService.getLists()
      .then(this.context.setListWishList)
      .catch(this.context.setError)
  }
  

  renderLists = () => {
    const { listWishList = [] } = this.context;
    return listWishList.map((list) => (
      TokenService.hasAuthToken() &&
      <ListWishList onDelete={this.refreshLists} key={list.id} list={list} />
    ));
  }

  renderAddButton = () => {
    if (TokenService.hasAuthToken())
       return (
        <Button className='Button__AddList' onClick={() => this.addListButtonHandler()}>
          + Add New List
        </Button>
      );
    else 
      return (
        <p className='AddPlaceHolder'>Please <Link className='signinLink' to='/login'>login</Link> or <Link className='signinLink' to='/register'>create an account</Link> to add a new list</p>
      );
  }

  render() {
    const { error } = this.context;
    return (
      <>
        <this.landingPageLogin/>
        <Section list className="WishListPage">
          {error ? (
            <p className="red">There was an error, try again</p>
          ) : (
            this.renderLists()
          )}
        </Section>
        <Section className='WishListPage__ButtonAndFormContainer'>
        <this.renderAddButton/>
        {this.state.add ? (
          <ListForm addedList={this.addedList} addList={this.addListsButtonHandler} />
        ) : null}
        </Section>
      </>
    );
  }
};
