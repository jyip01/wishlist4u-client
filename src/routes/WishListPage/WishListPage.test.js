import React from 'react';
import ReactDOM from 'react-dom';
import WishListPage from './WishListPage';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders without crashing',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><WishListPage/></BrowserRouter>,div)
    ReactDOM.unmountComponentAtNode(div)
})

it('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <WishListPage/>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
})
