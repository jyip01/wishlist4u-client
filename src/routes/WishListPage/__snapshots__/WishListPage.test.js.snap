// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
Array [
  <div
    className="WishListPage__Landing"
  >
    <h2
      className="WishListPage__Landing-header"
    >
      Welcome to WishList4U
    </h2>
    <p>
      Need help keeping track of all your wants and needs AS WELL as your friends and family? Look no further, WishList4U will help you keep track of all that!
    </p>
    <p>
      We've added some lists to get you started and give you an idea of how to use it, before adding your own list or making new wishes, please Login or Create an Account.
    </p>
    <p>
       If you're just here to demo the app please use the credentials below:
    </p>
    <ul
      className="WishListPage__Landing-demo-credentials"
    >
      <li>
        User Name: Test_User
      </li>
      <li>
        Password: P@ssword1
      </li>
    </ul>
  </div>,
  <section
    className="Section Section--list WishListPage"
  />,
  <section
    className="Section WishListPage__ButtonAndFormContainer"
  >
    <p
      className="AddPlaceHolder"
    >
      Please 
      <a
        className="signinLink"
        href="/login"
        onClick={[Function]}
      >
        login
      </a>
       or 
      <a
        className="signinLink"
        href="/register"
        onClick={[Function]}
      >
        create an account
      </a>
       to add a new list
    </p>
  </section>,
]
`;
